portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,56 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  static final String CONTENT_TYPE = "application/json;";
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Anthony!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List messages = new ArrayList<String>();
+    for (Entity entity : results.asIterable()){
+      String content = (String) entity.getProperty("content");
+      messages.add(content);
+    }
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+    response.setContentType(CONTENT_TYPE);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String content = request.getParameter("content").trim();
+    long timestamp = System.currentTimeMillis();
+    if (content.length() == 0) {
+      System.err.println("Empty comment submitted");
+      return;
+    }
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("content", content);
+    commentEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
```

Comment:
Not an immediate actionable item requested, but datastore service might be unavailable due to network conditions, service linkage etc. A failure will result in the servlet throwing an exception and showing a stacktrace to the response webpage, not so good looking as a web browsing experience. I would do a try {} catch {} for the specific exceptions, and return a friendly error message.

Reply:
Opened an issue: https://github.com/Anthony-Fiddes/portfolio/issues/13